.github/workflows/build.yml

```diff
@@ -0,0 +1,135 @@
+        default: 'dev'
+        options:
+          - dev
+          - prod
+
+# Read the target environment from workflow_dispatch inputs, or default to dev
+env:
+  ENVIRONMENT: ${{ inputs.ENVIRONMENT || 'dev' }}
+
+permissions:
+  contents: read
+
+jobs:
+  build-android:
+    runs-on: macos-latest
+    steps:
+      # Setup
+      - name: Convert environment to all caps
+        run: echo ENVIRONMENT_CAPS="$(echo "$ENVIRONMENT" | tr '[:lower:]' '[:upper:]')" >> "$GITHUB_ENV"
+      - name: Print environment
+        run: |
+          echo "Environment: $ENVIRONMENT ($ENVIRONMENT_CAPS)"
+      - uses: actions/checkout@v4.2.2
+        with:
+          persist-credentials: false
+
+      - uses: ./.github/actions/build-setup
+        with:
+          NPMRC_FILE: ${{ secrets.NPMRC_FILE }}
+          ENV_CONFIG_JS: ${{ vars[format('{0}_CONFIG_JS', env.ENVIRONMENT_CAPS)] }}
+          ENV_GOOGLE_SERVICES: ${{ vars[format('{0}_GOOGLE_SERVICES', env.ENVIRONMENT_CAPS)] }}
+
+      # Build the app
+      - name: Build the app
+        run: npm run build:app:android --env="$ENVIRONMENT"
+      - name: Rename build output
+        id: rename-output
+        run: |
+          mv "./platforms/android/app/build/outputs/apk/debug/app-debug.apk" "opal-${ENVIRONMENT}.apk"
+          echo "ARTIFACT_NAME=opal-${ENVIRONMENT}.apk" >> "$GITHUB_OUTPUT"
+      - name: Archive build output
+        uses: actions/upload-artifact@v4
+        with:
+          name: Android app
+          path: ${{ steps.rename-output.outputs.ARTIFACT_NAME }}
+
+  build-ios:
+    runs-on: macos-latest
+    steps:
+      # Setup
+      - name: Convert environment to all caps
+        run: echo ENVIRONMENT_CAPS="$(echo "$ENVIRONMENT" | tr '[:lower:]' '[:upper:]')" >> "$GITHUB_ENV"
+      - name: Print environment
+        run: |
+          echo "Environment: $ENVIRONMENT ($ENVIRONMENT_CAPS)"
+      - uses: actions/checkout@v4.2.2
+        with:
+          persist-credentials: false
+
+      - uses: ./.github/actions/build-setup
+        with:
+          NPMRC_FILE: ${{ secrets.NPMRC_FILE }}
+          ENV_CONFIG_JS: ${{ vars[format('{0}_CONFIG_JS', env.ENVIRONMENT_CAPS)] }}
+          ENV_GOOGLE_SERVICES: ${{ vars[format('{0}_GOOGLE_SERVICES', env.ENVIRONMENT_CAPS)] }}
+
+      # Install an Apple certificate and provisioning profile used to build the app for iOS
+      # See: https://docs.github.com/en/actions/use-cases-and-examples/deploying/installing-an-apple-certificate-on-macos-runners-for-xcode-development
+      - name: Install the Apple certificate and provisioning profile
+        env:
+          BUILD_CERTIFICATE_BASE64: ${{ secrets.BUILD_CERTIFICATE_BASE64_FILE }}
+          P12_PASSWORD: ${{ secrets.BUILD_CERTIFICATE_PASSWORD }}
+          BUILD_PROVISION_PROFILE_BASE64: ${{ secrets[format('{0}_PROVISIONING_PROFILE_BASE64_FILE', env.ENVIRONMENT_CAPS)] }} # zizmor: ignore[overprovisioned-secrets]
+          KEYCHAIN_PASSWORD: ${{ secrets.CREATE_KEYCHAIN_PASSWORD }}
```

Comment:
nit: Can we name the secret `KEYCHAIN_PASSWORD` instead?

Reply:
I was trying to imply with the name that this is a new thing we're creating, so the password can be anything. But I agree, it could be clearer. Any ideas? `TEMPORARY_KEYCHAIN_PASSWORD`?